RecoParticleFlow/PFClusterProducer/src/Cluster3DPCACalculator.cc

```diff
@@ -0,0 +1,132 @@
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Utilities/interface/isFinite.h"
+
+#include <cmath>
+#include <unordered_map>
+
+#include "vdt/vdtMath.h"
+
+#include "RecoParticleFlow/PFClusterProducer/interface/PFCPositionCalculatorBase.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecHitFwd.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecHit.h"
+
+#include "DataFormats/ForwardDetId/interface/HGCalDetId.h"
+
+#include "TPrincipal.h"
```

Comment:
@Dr15Jones 
Is TPrincipal thread safe?
It appears to be the first time use of it in CMSSW compiled code.